src/classes/BGE_BatchGiftEntry_UTIL.cls

```diff
@@ -51,7 +51,41 @@ public with sharing class BGE_BatchGiftEntry_UTIL {
     * @return List<String> list of DataImport__c field api names
     */
     public static List<String> getDataImportFields(Id batchId) {
-        return getDataImportFields(batchId, true);
+        List<String> fieldNames = new List<String>();
```

Comment:
I tried to find the best place to introduce the new path with minimal effect to BGE code, but it might be smart to introduce a new method entirely.  I'll take another look.

Reply:
Just curious also - if this maintains both without too much impact, I'm happy to stick with a singular method. It does make the code base a little cleaner without a second method!

Reply:
Yeah, and looking a little deeper again, I think this method is probably the best place to perform the check against the type of batch and return either the fields from the Active_Fields or from Advanced Mapping.  Callers to this method don't need to know which batch type it is and should get the correct set of fields back either way.  On a larger refactor we could look at how the data import and batch information are sent from the component to Apex and how/when they are queried (and re-queried) throughout this method chain, but I think this is a good path for now.